p2p/tests/sync.rs

```diff
@@ -193,8 +193,49 @@ where
     T: NetworkingService,
     T::SyncingCodecHandle: SyncingCodecService<T>,
 {
-    let event = mgr.handle_mut().poll_next().await.unwrap();
-    mgr.on_syncing_event(event).await.unwrap();
+    match mgr.handle_mut().poll_next().await.unwrap() {
+        net::types::SyncingEvent::Request {
+            peer_id,
+            request_id,
+            request:
+                Message {
+                    msg: MessageType::Syncing(SyncingMessage::Request(message)),
+                    magic: _,
+                },
+        } => match message {
+            SyncingRequest::GetHeaders { locator } => {
+                mgr.process_header_request(peer_id, request_id, locator).await?;
+            }
+            SyncingRequest::GetBlocks { block_ids } => {
+                mgr.process_block_request(peer_id, request_id, block_ids).await?;
+            }
+        },
+        net::types::SyncingEvent::Response {
+            peer_id,
+            request_id: _,
+            response:
+                Message {
+                    msg: MessageType::Syncing(SyncingMessage::Response(message)),
+                    magic: _,
+                },
+        } => match message {
+            SyncingResponse::Headers { headers } => {
+                mgr.process_header_response(peer_id, headers).await?;
+            }
+            SyncingResponse::Blocks { blocks } => {
+                mgr.process_block_response(peer_id, blocks).await?;
+            }
+        },
+        net::types::SyncingEvent::Error {
+            peer_id,
+            request_id,
+            error,
+        } => {
+            mgr.process_error(peer_id, request_id, error).await?;
+        }
+        _ => {}
```

Comment:
Is this necessary? I thought the arms are exhaustive.

Reply:
It was not, the invalid messages are ignored for now. If I write more comprehensive tests in `tests/sync.rs` I'll match against those too but I'll leave this as is for now.